src/vs/workbench/contrib/customEditor/browser/extensionContributedEditorService.ts

```diff
@@ -0,0 +1,351 @@
+		priority: number,
+		editorInfo: ContributedEditorInfo,
+		options: ContributionPointOptions,
+		createEditorInput: EditorInputFactoryFunction,
+		createDiffEditorInput?: DiffEditorInputFactoryFunction
+	): IDisposable {
+		if (this._contributionPoints.get(globPattern) === undefined) {
+			this._contributionPoints.set(globPattern, []);
+		}
+		const remove = insert(this._contributionPoints.get(globPattern)!, {
+			globPattern,
+			priority,
+			editorInfo,
+			options,
+			createEditorInput,
+			createDiffEditorInput
+		});
+		return toDisposable(() => remove());
+	}
+
+	contributedEditorOverride(handler: IExtensionContributedEditorHandler): IDisposable {
+		const remove = insert(this.extensionContributedEditors, handler);
+		return toDisposable(() => remove());
+	}
+
+	hasContributionPoint(schemeOrGlob: string): boolean {
+		return this._contributionPoints.has(schemeOrGlob);
+	}
+
+	getAssociationsForResource(resource: URI): EditorAssociations {
+		const rawAssociations = this.configurationService.getValue<EditorAssociations>(editorsAssociationsSettingId) || [];
+		return rawAssociations.filter(association => CustomEditorInfo.selectorMatches(association, resource));
+	}
+
+	private findMatchingContributions(resource: URI): ContributionPoint[] {
+		let contributions: ContributionPoint[] = [];
+		// Then all glob patterns
+		for (const key of this._contributionPoints.keys()) {
+			const contributionPoints = this._contributionPoints.get(key)!;
+			for (const contributionPoint of contributionPoints) {
+				const matchOnPath = typeof contributionPoint.globPattern === 'string' && contributionPoint?.globPattern.indexOf(posix.sep) >= 0;
+				const target = matchOnPath ? resource.path : basename(resource);
+				if (glob.match(contributionPoint.globPattern, target)) {
+					contributions.push(contributionPoint);
+				}
+			}
+		}
+		// Return the contributions sorted by their priority
+		return contributions.sort((a, b) => b.priority - a.priority);
+	}
+
+	private getContributionPoint(resource: URI, override: string | undefined) {
+		let contributionPoints = this.findMatchingContributions(resource);
+		if (override) {
+			return contributionPoints.find(contribPoint => contribPoint.editorInfo.id === override);
+		}
+
+		const associationsFromSetting = this.getAssociationsForResource(resource);
+		// We only want built-in+ if no user defined setting is found. Else we will fall back to the text editor
+		const contributionPoint = contributionPoints.find(contribPoint => contribPoint.priority >= priorityToRank(ContributedEditorPriority.builtin));
+		// If the user has a setting we use that, else choise the highest priority editor that is built-in+
+		const selectedViewType = associationsFromSetting[0]?.viewType || contributionPoint?.editorInfo.id;
+
+		if (associationsFromSetting.length === 0 && contributionPoint?.editorInfo.priority === ContributedEditorPriority.default) {
+			setTimeout(() => {
+				console.log('Conflicting defaults!');
+			}, 500);
```

Comment:
:eyes:

Reply:
This was my testing, after this PR we want to prompt users when there's conflicting defaults to choose a default going forward. I will remove this.